presto-main/src/main/java/com/facebook/presto/sql/planner/StatisticsAggregationPlanner.java

```diff
@@ -110,26 +110,49 @@ public TableStatisticAggregation createStatisticsAggregation(TableStatisticsMeta
         return new TableStatisticAggregation(aggregation, descriptor.build());
     }
 
-    private ColumnStatisticsAggregation createColumnAggregation(ColumnStatisticType statisticType, VariableReferenceExpression input)
+    private ColumnStatisticsAggregation createColumnAggregation(ColumnStatisticType statisticType, VariableReferenceExpression input, boolean afterTranslateExpressions)
     {
-        SymbolReference symbolReference = new SymbolReference(input.getName());
-        switch (statisticType) {
-            case MIN_VALUE:
-                return createAggregation("min", symbolReference, input.getType(), input.getType());
-            case MAX_VALUE:
-                return createAggregation("max", symbolReference, input.getType(), input.getType());
-            case NUMBER_OF_DISTINCT_VALUES:
-                return createAggregation("approx_distinct", symbolReference, input.getType(), BIGINT);
-            case NUMBER_OF_NON_NULL_VALUES:
-                return createAggregation("count", symbolReference, input.getType(), BIGINT);
-            case NUMBER_OF_TRUE_VALUES:
-                return createAggregation("count_if", symbolReference, BOOLEAN, BIGINT);
-            case TOTAL_SIZE_IN_BYTES:
-                return createAggregation(SumDataSizeForStats.NAME, symbolReference, input.getType(), BIGINT);
-            case MAX_VALUE_SIZE_IN_BYTES:
-                return createAggregation(MaxDataSizeForStats.NAME, symbolReference, input.getType(), BIGINT);
-            default:
-                throw new IllegalArgumentException("Unsupported statistic type: " + statisticType);
+        if (!afterTranslateExpressions) {
+            SymbolReference symbolReference = new SymbolReference(input.getName());
```

Comment:
@pguofb Please try to rewrite this code to avoid code duplication.

Let's keep only a single version of the `createColumnAggregation` and the `createAggregation`. Let's change the type of the input to `RowExpression`  for both and let's pass either `VariableReferenceExpression` or `OriginalExpression` based on the flag from the `createStatisticsAggregation`. Also let's change the flag name to something like `useOriginalExpression` and set it to `true` in the `LogicalPlanner`. Also please add a comment that this is transitional, and must be migrated to `RowExpression` after these are fully supported by all the planner rules.